test/libyul/yulOptimizerTests/fullSuite/unusedFunctionParameterPruner_loop.yul

```diff
@@ -20,18 +20,46 @@
 // {
 //     {
 //         f()
-//         f()
-//         f()
+//         f_72()
+//         f_73()
 //         sstore(0, 1)
 //     }
 //     function f()
 //     {
+//         let a := 1
+//         let b := 10
+//         let a_1 := calldataload(0)
+//         let _1 := iszero(a_1)
+//         for { } iszero(b) { b := add(b, not(0)) }
+//         {
+//             a := a_1
+//             mstore(a_1, 0)
+//             if _1 { leave }
+//         }
+//     }
+//     function f_72()
+//     {
+//         let a := 2
+//         let b := 10
+//         let a_1 := calldataload(0)
+//         let _1 := iszero(a_1)
+//         for { } iszero(b) { b := add(b, not(0)) }
+//         {
+//             a := a_1
+//             mstore(a_1, 0)
+//             if _1 { leave }
+//         }
+//     }
+//     function f_73()
+//     {
+//         let a := 3
 //         let b := 10
-//         let a := calldataload(0)
-//         let _1 := iszero(a)
+//         let a_1 := calldataload(0)
+//         let _1 := iszero(a_1)
 //         for { } iszero(b) { b := add(b, not(0)) }
 //         {
-//             mstore(a, 0)
+//             a := a_1
+//             mstore(a_1, 0)
 //             if _1 { leave }
 //         }
 //     }
```

Comment:
Looks like the changes show up not only as gas differences. Two tests also got updated output.

In this case it looks like a clear downgrade. The optimizer is definitely leaving some possible optimizations on the table. Still, not sure how common this is in practice. I'm going to play with this when tweaking the optimizer sequence.

Reply:
I mean, the optimizer is mainly being rather stupid here - it specializes all three calls and then trusts that since the arguments are unused, the result will collapse to identical functions that can be deduplicated... the simpler choice would have been to not specialize in the first place.
For cases like this it's not worth preserving optimizations that work out by chance. Even if someone thought it funny to include it as test case :-).